src/d3-ext/profile.js

```diff
@@ -0,0 +1,316 @@
+/**
+ * @fileoverview Provides a D3js component to be used to draw an elevation
+ * profile chart.
+ *
+ * Example usage:
+ *
+ * var selection = d3.select('#element_id');
+ * var profile = ngeo.profile().onHover(function(point) {
+ *   console.log(point.x, point.y);
+ * }).onOut(function() {console.log("out");});
+ * selection.datum(data).call(profile);
+ *
+ *
+ * The selection data to be used should look like the following:
+ *
+ * [
+ *     {
+ *         "y": 199340,
+ *         "values": {"mnt": 788.7},
+ *         "dist": 0.0,
+ *         "x": 541620
+ *     }, ...
+ * ]
+ *
+ */
+goog.provide('ngeo.profile');
+
+
+/**
+ *
+ * @return {Object}
+ * @export
+ *
+ */
+ngeo.profile = function() {
+
+  /**
+   * The values for margins around the chart defined in pixels.
+   */
+  var margin = {top: 10, right: 10, bottom: 30, left: 40};
+
+  /**
+   * Method to get the coordinate in pixels from a distance.
+   */
+  var bisectDistance = d3.bisector(function(d) { return d['dist']; }).left;
+
+  /**
+   * Distance units. Either 'm' or 'km'.
+   */
+  var units;
+
+  /**
+   * Factor to determine whether to use 'm' or 'km'.
+   */
+  var xFactor;
+
+  /**
+   * @type {function(Object)}
+   * Hover callback function.
+   */
+  var hoverCallback = goog.nullFunction;
+
+  /**
+   * @type {function()}
+   * Out callback function.
+   */
+  var outCallback = goog.nullFunction;
+
+  /**
+   * The color to be used for filling the area.
+   * Can be overriden using the '.area' CSS selector.
+   * @type {string}
+   */
+  var FILL_COLOR = 'rgba(222, 222, 222, 0.5)';
```

Comment:
Why capital letters? Because it's a constant? You may want to use a `@const` annotation.